src/App.tsx

```diff
@@ -1,24 +1,291 @@
+      id: 0,
+      userId: USER_ID,
+      title: title.trim(),
+      completed: false,
+    };
+
+    setTempTodo(createdTodo);
+
+    todoService.createTodo({ userId, title, completed })
+      .then(newTodo => {
+        setTodos(currentTodos => [...currentTodos, newTodo]);
+        setTitleTodo('');
+      })
+      .catch(() => {
+        setErrorMessage(Message.NoAddTodo);
+      })
+      .finally(() => {
+        setIsLoading(false);
+        setTempTodo(null);
+      });
+  };
+
+  const removeComplatedTodos = (todosId: number[] | null) => {
+    setIsLoading(true);
+
+    if (todosId) {
+      setDeletedTodo(todosId);
+      todosId.map(todoId => {
+        return todoService.deleteTodos(todoId)
+          .then(() => {
+            setTodos(currentTodos => (
+              currentTodos.filter(todo => todo.id !== todoId)
+            ));
+          })
+          .catch((error) => {
+            setDeletedTodo(null);
+            setErrorMessage(Message.NoDeleteTodo);
+            throw error;
+          })
+          .finally(() => {
+            setIsLoading(false);
+            setDeletedTodo(null);
+          });
+      });
+    }
+  };
+
+  const updateTodoStatus = (updatedTodo: Todo) => {
+    setErrorMessage('');
+    setIsLoading(true);
+    setChangedTodo([updatedTodo.id]);
+
+    return todoService.updateTodo(updatedTodo)
+      .then(() => {
+        setTodos(todos.map(todo => (
+          todo.id === updatedTodo.id
+            ? {
+              ...todo,
+              completed: updatedTodo.completed,
+
+            }
+            : todo
+        )));
+      })
+
+      .catch(() => setErrorMessage(Message.NoUpdateTodo))
+      .finally(() => {
+        setIsLoading(false);
+        setChangedTodo(null);
+      });
+  };
+
+  const updateTodosAllStatus = (updatedTodos: Todo[]) => {
+    setErrorMessage('');
+    setIsLoading(true);
+    const changedTodoId: number[] = [];
+
+    updatedTodos.forEach(updatedTodo => {
+      const similarTodo = todos.find(todo => todo.id === updatedTodo.id);
+
+      if (similarTodo && similarTodo.completed !== updatedTodo.completed) {
+        changedTodoId.push(similarTodo.id);
+      }
+    });
+
+    setChangedTodo(changedTodoId);
+
+    const updatedTodoSirvice = updatedTodos.map(async (todo) => {
+      try {
+        const todoChange = await todoService.updateTodo(todo);
+
+        return todoChange;
+      } catch (error) {
+        setErrorMessage(Message.NoUpdateTodo);
+
+        return todo;
+      }
+    });
+
+    Promise.all(updatedTodoSirvice)
+      .then((gettedTodos) => {
+        const gettedTodosId = gettedTodos.map(t => t.id);
+
+        setTodos(curT => curT.map(
+          item => (
+            gettedTodosId.includes(item.id)
+              ? { ...item, completed: gettedTodos[0].completed }
+              : item
+          ),
+        ));
+      })
+      .finally(() => {
+        setIsLoading(false);
+        setChangedTodo(null);
+      });
+  };
```

Comment:
this function should be refactored and simplified because it is hard to read. pay attention to naming